src/main/java/org/terasology/logic/health/event/OnRestoredEvent.java

```diff
@@ -17,14 +17,29 @@
 
 import org.terasology.entitySystem.entity.EntityRef;
 
+/**
+ * This event is sent after the entity is restored. Final event of restoration logic.
+ */
 public class OnRestoredEvent extends HealthChangedEvent {
+    /**
+     * The amount by which the entity is restored.
+     */
     private int amount;
 
+    /**
+     * Constructor for creating the event.
```

Comment:
Another edge case: should plain constructors have javadoc? In this case maybe if the wording is a slight bit more meaningful (we already know by the constructors presence that it will create the event, so the text doesn't really add anything). I'm not sure if there's really any other meaning to add in this case.